verifier/src/main/kotlin/tech/figure/classification/asset/verifier/client/VerifierClient.kt

```diff
@@ -87,18 +89,28 @@ class VerifierClient(private val config: VerifierClientConfig) {
         val currentHeight = config.eventStreamProvider.currentHeight()
         var latestBlock = startingBlockHeight?.takeIf { start -> start > 0 && currentHeight?.let { it >= start } != false }
 
+        suspend fun newBlock(height: Long) {
+            // Record each block intercepted
+            NewBlockReceived(height).send()
+            // Track new block height
+            latestBlock = trackBlockHeight(latestBlock, height)
+        }
+
         config.eventStreamProvider.startProcessingFromHeight(
             latestBlock,
             onBlock = { block ->
-                // Record each block intercepted
-                NewBlockReceived(block).send()
-                // Track new block height
-                latestBlock = trackBlockHeight(latestBlock, block.height)
+                when (block) {
```

Comment:
Forcing compatible types into the `VerifierClient` will make this framework inflexible if a third or custom implementation exists elsewhere.  I think the best bet is to instead maybe call this `recordBlockHeight` and just have the height itself passed in.  Then you don't have to worry about types.

Reply:
Also, I think each interface should be required to implement a function, like `fun toAssetClassificationEvent(data: T)`.  Then we wouldn't have to have the additional block api OR event stream types in the `AssetClassificationEvent` class itself, and the event stream provider can just inherently understand how to process the "block data thing"

Reply:
Last note: After seeing this implementation, it seems like you can honestly get away with not even having a generic capture for block type in the `EventStreamProvider` interface, contrary to what I said earlier.  If you change to `recordBlockHeight`, each provider its can fully control its transformations and block data.  As long as each provider can emit `AssetClassificationEvent` objects, it shouldn't matter how its internals interpret block information or where it comes from

Reply:
I think i've addressed all of this in the latest commit. Take a look at let me know if you have any concerns still.